src/components/cart.js

```diff
@@ -220,21 +237,57 @@ export default class Cart extends Component {
    */
   setQuantity(target, fn) {
     const id = target.getAttribute('data-line-item-id');
-    const item = this.model.lineItems.filter((lineItem) => lineItem.id === id)[0];
+    const item = this.model.lineItems.find((lineItem) => lineItem.id === id);
     const newQty = fn(item.quantity);
     return this.props.tracker.trackMethod(this.updateItem.bind(this), 'Update Cart', this.cartItemTrackingInfo(item, newQty))(id, newQty);
   }
 
   /**
+   * set cache using line items.
+   * @param {Array} lineItems - array of GraphModel line item objects.
+   */
+  updateCache(lineItems) {
+    if (this.cache) {
+      this.cache = lineItems.map((item) => {
+        const updatedItem = this.cache.find((cacheItem) => {
+          return cacheItem.id === item.id;
+        });
+        return Object.assign({}, updatedItem, item);
+      });
+    } else {
+      this.cache = lineItems.map((item) => {
+        return Object.assign({}, item);
+      });
+    }
```

Comment:
I'm pretty sure you could eliminate the branching here.

first off: if you assign `this.cache = []` in the constructor, instead of null, you can just use the code in the `if` part, and clear out the `else` part.

Also, find does a full iteration of the cache for every pass of the line items. Since this is a performance PR, the code below does at most `lineItems.lengthh * 2` iterations:

```javascript
const lineItemCache = this.cache.reduce((acc, item) => {
  acc[item.id] = item;

  return acc;
}, {});

this.cache = lineItems.map((item) => {
  return Object.assign({}, lineItemCache[item.id], item);
});
```